### PR TITLE
split static condition tests from net_match_config()

### PR DESCRIPTION
**Pull request by yuwata**
Strictly speaking, this breaks backward compatibility of conditions in `[Match]` section; E.g.,
```
[Match]
Host=hogehoge
Host=foover
```
Previously, the setting `Host=hogehoge` was ignored, and the last entry was always used. This makes all conditions are examined, and the results are ANDed. So, no host cannot match the above condition.

To override the preexisting conditions, empty string assignment is necessary, that is,
```
[Match]
Host=hogehoge
Host=
Host=foover
```
Then the condition `Host=hogehoge` is cleared and only `Host=foover` is used.

---
The backward incompatibility is dropped from this PR.

### CODE REVIEW

File: man/systemd.link.xml [hunk 1]
```diff
@@ -119,7 +119,9 @@
           <para>Matches against the hostname or machine
           ID of the host. See <varname>ConditionHost=</varname> in
           <citerefentry><refentrytitle>systemd.unit</refentrytitle><manvolnum>5</manvolnum></citerefentry>
-          for details.</para>
+          for details. When prefixed with an exclamation mark (<literal>!</literal>), the result is nagated.
```
Comment (keszybz) [inline man/systemd.link.xml]: "negated"

File: man/systemd.link.xml [hunk 2]
```diff
@@ -130,7 +132,9 @@
           whether it is a specific implementation. See
           <varname>ConditionVirtualization=</varname> in
           <citerefentry><refentrytitle>systemd.unit</refentrytitle><manvolnum>5</manvolnum></citerefentry>
-          for details.</para>
+          for details. When prefixed with an exclamation mark (<literal>!</literal>), the result is nagated.
```
Comment (keszybz) [inline man/systemd.link.xml]: "negated"

File: man/systemd.link.xml [hunk 3]
```diff
@@ -140,28 +144,32 @@
...
-          <listitem>
-            <para>Checks whether the kernel version (as reported by <command>uname -r</command>) matches a certain
-            expression (or if prefixed with the exclamation mark does not match it). See
-            <varname>ConditionKernelVersion=</varname> in
-            <citerefentry><refentrytitle>systemd.unit</refentrytitle><manvolnum>5</manvolnum></citerefentry> for
-            details.
-            </para>
-          </listitem>
-        </varlistentry>
+          for details. When prefixed with an exclamation mark (<literal>!</literal>), the result is nagated.
```
Comment (keszybz) [inline man/systemd.link.xml]: "negated"

File: man/systemd.link.xml [hunk 4]
```diff
@@ -140,28 +144,32 @@
...
+          When specified more than once, the results are ANDed. If an empty string is assigned,
+          then previously assigned values are cleared.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>KernelVersion=</varname></term>
+        <listitem>
+          <para>Checks whether the kernel version (as reported by <command>uname -r</command>) matches a certain
+          expression (or if prefixed with the exclamation mark does not match it). See
```
Comment (keszybz) [inline man/systemd.link.xml]: This information about negation is now repeated needlessly. I think the part in parentheses should be removed.

File: man/systemd.link.xml [hunk 5]
```diff
@@ -140,28 +144,32 @@
...
+          <varname>ConditionKernelVersion=</varname> in
+          <citerefentry><refentrytitle>systemd.unit</refentrytitle><manvolnum>5</manvolnum></citerefentry> for
+          details. When prefixed with an exclamation mark (<literal>!</literal>), the result is nagated.
```
Comment (keszybz) [inline man/systemd.link.xml]: "negated"

File: man/systemd.link.xml [hunk 6]
```diff
@@ -140,28 +144,32 @@
...
       <varlistentry>
         <term><varname>Architecture=</varname></term>
         <listitem>
           <para>Checks whether the system is running on a specific
-          architecture. See <varname>ConditionArchitecture=</varname>
-          in
+          architecture. See <varname>ConditionArchitecture=</varname> in
           <citerefentry><refentrytitle>systemd.unit</refentrytitle><manvolnum>5</manvolnum></citerefentry>
-          for details.</para>
+          for details. When prefixed with an exclamation mark (<literal>!</literal>), the result is nagated.
```
Comment (keszybz) [inline man/systemd.link.xml]: "negated", here and below.

**Comment by yuwata** [comment]
Thanks 😊 I will fix them soon.

**Comment by yuwata** [comment]
> I think we can do this. But if multiple values are assigned, I think we should at least emit a warning. It's very unlikely that this is intentional.

OK. I did not think it was necessary, but this breaks a backward compatibility, so it is reasonable.

**Comment by keszybz** [comment]
So this should go in first, then #11861, and then #11893. So I'll wait with reviews of the other ones for the update here.

**Comment by yuwata** [comment]
Updated.

> I think we can do this. But if multiple values are assigned, I think we should at least emit a warning. It's very unlikely that this is intentional.

After re-considering this, if a unit has multiple conditions and that is intentional, then the warning is noisy and may confuse people...
I think we should warn only when the condition is always true or always false, e.g.,
```
Host=hoge
Host=foo
```
But implementing such a check is out of the scope of this PR. So, I've dropped the multiple assignment support, and I'd like to leave that for another PR. Though, I am not sure people really wants such a feature...

**Comment by keszybz** [comment]
Needs a rebase now.

**Comment by yuwata** [comment]
@keszybz Thank you for the review. Rebased. PTAL.

**Pull request merged**